OAK-10141: relevance on queries involving dynamic boost sometimes differ from lucene

fabriziofortino opened the pull request:
* refactoring of full-text tests to share them between lucene and elastic
* fix an issue on elastic where full-text queries wrongly assigned the boost factor on specific fields
* fix relevance in Elastic when dynamic boost is enabled. The dynamic terms need to be de-boosted (as it happens in lucene). To properly address the issue, a new field has been created. This is not a breaking change since the queries won't complain in case the field is not yet there. A reindex is required to get the correct relevant results.

fabriziofortino committed:
OAK-10111: (refactor) rename FullTextIndexCommonTest to FullTextAnalyzerCommonTest

fabriziofortino committed:
OAK-10141: (refactor) move elastic fulltext tests in oak-search

fabriziofortino committed:
OAK-10141: correctly assign boosts on fulltext queries

fabriziofortino committed:
OAK-10141: refactored dynamic boost tests to be shared between lucene and elastic

fabriziofortino committed:
OAK-10141: de-boost dynamic boost terms in Elastic to align relevance with Lucene

fabriziofortino committed:
OAK-10141: (minor) fixed indentation

fabriziofortino committed:
Merge branch 'trunk' into OAK-10141

nit0906 reviewed (approved):
lgtm - If possible,  can we check that we didn't miss any test during the refactoring ? 

Maybe just compare the list of tests from a build on trunk vs this ?

fabriziofortino commented:
@nit0906 the list of tests is not the same as before. While refactoring, I noticed duplicate tests that I removed and added new tests. To easily see what was done in the refactoring, you can have a look at the specific commits in this PR. Every commit has a clear description and you can see the relevant changes in smaller chunks.
